book/art/fullnode-proposal.bob

```diff
@@ -0,0 +1,30 @@
+             .----------------------------------.
+             |  Fullnode                        |
+             |                                  |
+ .--------.  |  .------------------.            |
+ |        |---->|                  |            |
+ | Client |  |  | JSON RPC Service |            |
+ |        |<----|                  |            |
+ `----+---`  |  `------------------`            |
+      |      |     ^                            |
+      |      |     |    .----------------.      | .------------------.
+      |      |     |    | Gossip Service |<------>| Validators       |
+      |      |     |    `----------------`      | |                  |
+      |      |     |         ^                  | |                  |
+      |      |     |         |                  | |  .------------.  |
+      |      |  .--+---.  .--+---.  .---------. | |  |            |  |
+      |      |  | Bank |<-|Replay|  | Peer TX |<-----+ Upstream   |  |
+      |      |  | Forks|  |Stage |  | Receiver| | |  | Validators |  |
+      |      |  `------`  `------`  `-+-------` | |  |            |  |
+      |      |     ^            ^     |         | |  `------------`  |
+      |      |     |            |     v         | |                  |
+      |      |     |          .-+-------.       | |                  |
+      |      |     |          |Blocktree|       | |                  |
```

Comment:
I don't care for that notification from Blocktree to ReplayStage.  I've been down the reactive programming rabbit hole enough times in my career to fear it. Passive data structures are just so much simpler. How about a line from the previous stage that represents messages with a fork id and slot height? That way Blocktree can remain passive and simply be queried by ReplayStage.

In other words, the current architecture is to send entries from RetransmitStage to ReplayStage. Instead, write them to Blocktree and tell ReplayStage where to find those entries.

Reply:
Looks like the current code already does what's described in the proposal w.r.t. notifying the ReplayStage from Blocktree. @carllin